core/lib/zksync_core/src/native_token_fetcher/mod.rs

```diff
@@ -74,11 +77,21 @@ impl NativeTokenFetcher {
 
 #[async_trait]
 impl ConversionRateFetcher for NativeTokenFetcher {
-    fn conversion_rate(&self) -> anyhow::Result<u64> {
-        anyhow::Ok(
-            self.latest_to_eth_conversion_rate
-                .load(std::sync::atomic::Ordering::Relaxed),
-        )
+    fn conversion_rate(&self) -> anyhow::Result<BigDecimal> {
+        let lock = match self.latest_to_eth_conversion_rate.lock() {
+            Ok(lock) => lock,
+            Err(err) => {
+                tracing::error!(
+                    "Error while getting lock of latest conversion rate: {:?}",
+                    err,
+                );
+                return Err(anyhow!(
+                    "Error while getting lock of latest conversion rate: {:?}",
+                    err,
+                ));
+            }
+        };
```

Comment:
Failing to get the lock can be handled in two ways:
- (current) Generate trace error and terminate program by propagating `Err`.
- (proposed) Request conversion rate to component, wrap it inside a new `Arc/Mutex` and set is as new `latest_to_eth_conversion_rate`.